api/appStore/deployment/AppStoreDeploymentRestHandler.go

```diff
@@ -507,3 +511,91 @@ func (handler AppStoreDeploymentRestHandlerImpl) GetInstalledAppVersion(w http.R
 
 	common.WriteJsonResp(w, err, dto, http.StatusOK)
 }
+
+func (handler AppStoreDeploymentRestHandlerImpl) UpdateProjectHelmApp(w http.ResponseWriter, r *http.Request) {
+
+	userId, err := handler.userAuthService.GetLoggedInUser(r)
+	if userId == 0 || err != nil {
+		common.WriteJsonResp(w, err, "Unauthorized User", http.StatusUnauthorized)
+		return
+	}
+
+	token := r.Header.Get("token")
+
+	var request appStoreBean.UpdateProjectHelmAppDTO
+
+	decoder := json.NewDecoder(r.Body)
+
+	err = decoder.Decode(&request)
+
+	if err != nil {
+		common.WriteJsonResp(w, err, nil, http.StatusBadRequest)
+		return
+	}
+
+	request.UserId = userId
+
+	if request.InstalledAppId == 0 {
+		appIdentifier, err := handler.helmAppService.DecodeAppId(request.AppId)
+		if err != nil {
+			handler.Logger.Errorw("error in decoding app id", "err", err)
+			common.WriteJsonResp(w, err, "error in decoding app id", http.StatusBadRequest)
+		}
+
+		// this rbac object checks that whether user have permission to change current project.
+
+		rbacObjectForCurrentProject, rbacObjectForCurrentProject2 := handler.enforcerUtilHelm.GetHelmObject(appIdentifier.ClusterId, appIdentifier.Namespace, appIdentifier.ReleaseName)
+
+		ok := handler.enforcer.Enforce(token, casbin.ResourceHelmApp, casbin.ActionUpdate, rbacObjectForCurrentProject) || handler.enforcer.Enforce(token, casbin.ResourceHelmApp, casbin.ActionUpdate, rbacObjectForCurrentProject2)
```

Comment:
code from here till end of condition in if and else is duplicated. you are just getting different rbac objects for first enforcer on the basis of installedAppId. extract rest of the code to remove duplicacy.